lib/rule-context.js

```diff
@@ -70,14 +70,18 @@ var PASSTHROUGHS = [
  * @param {array} options The configuration information to be added to the rule.
  * @param {object} settings The configuration settings passed from the config file.
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
+ * @param {string} parser The parser setting passed from the config file.
  */
-function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures, parser) {
```

Comment:
I was torn, wasn't sure if it should match the original config property name (`parser`) or the RuleContext property (`parserName`). Went with the former, though with both provided as args, the latter makes more sense.